xdsl/ir.py

```diff
@@ -1006,6 +1016,16 @@ def add_ops(self, ops: Iterable[Operation]) -> None:
         for op in ops:
             self.add_op(op)
 
+    def insert_ops_before(self, ops: list[Operation],
+                          existing_op: Operation) -> None:
+        index = self.get_operation_index(existing_op)
+        self.insert_op(ops, index)
+
+    def insert_ops_after(self, ops: list[Operation],
+                         existing_op: Operation) -> None:
+        index = self.get_operation_index(existing_op)
+        self.insert_op(ops, index + 1)
+
     def insert_op(self,
```

Comment:
Why the split into `insert_op` and `insert_ops`? I agree that it is better naming, but this doesn't seem to be something that `insert_op` does, so why do it now?

Reply:
Yeah this is leaking changes from the future, where inserting ops will run insert op in a loop, since there will be things like updating next/previous to do

Reply:
Okay sure, if you know the future, I am fine with merging this.

Reply:
🔮

Reply:
Why should we have two functions for this?
IMO we should not treat differently something that only differs in size.

Reply:
I don't feel super strongly about this so I removed the singular version